srcCpp/Infrastructure_pro.cxx

```diff
@@ -537,9 +537,12 @@ bool configureShmemTransport(
     if (parentProp != NULL && parentProp->value != NULL) {
         parentMsgSizeMax = atoi(parentProp->value);
         messageSizeMaxSet = true;
-    } else if (qos.transport_builtin.mask == DDS_TRANSPORTBUILTIN_SHMEM) {
-        if ((datalen + MESSAGE_OVERHEAD_BYTES) > DEFAULT_MESSAGE_SIZE_MAX) {
-            parentMsgSizeMax = datalen + MESSAGE_OVERHEAD_BYTES;
+    } else {
+        if (qos.transport_builtin.mask == DDS_TRANSPORTBUILTIN_SHMEM) {
+            if ((datalen + MESSAGE_OVERHEAD_BYTES) > parentMsgSizeMax) {
+                parentMsgSizeMax = datalen + MESSAGE_OVERHEAD_BYTES;
+                transport.minimumMessageSizeMax = parentMsgSizeMax;
+            }
         }
```

Comment:
ok. I agree with this part so far. Just to clarify though:

If you are setting the property for SHMEM but it is SHMEM + something, then you use the value of the property. If you are NOT setting the property and it is SHMEM + something, then you use whatever is the minimum for the max_size across all transports.

That kind of makes sense, but it is also a weird scenario.

Reply:
If you set the property "dds.transport.shmem.builtin.parent.message_size_max" for SHMEM we always use the property. 

If you don't set the property we only set it if the only transport is SHMEM

Reply:
That makes sense to me.